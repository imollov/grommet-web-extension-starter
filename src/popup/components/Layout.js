import React from 'react';
import PropTypes from 'prop-types';
import { Box, Grommet } from 'grommet';
import { grommet } from 'grommet/themes';

const Layout = ({ ...rest }) => (
  <Grommet theme={grommet}>
    <Box width={{ min: '350px' }} height={{ min: '350px' }} {...rest} />
  </Grommet>
);

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Layout;
