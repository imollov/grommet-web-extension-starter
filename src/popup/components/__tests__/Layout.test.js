import React from 'react';
import { shallow } from 'enzyme';

import Layout from '../Layout';

describe('Layout Component', () => {
  const children = <div>Test</div>;

  it('smoke test', () => {
    const wrapper = shallow(<Layout>{children}</Layout>);

    expect(wrapper).toExist();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Layout>{children}</Layout>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should contain children', () => {
    const wrapper = shallow(<Layout>{children}</Layout>);

    expect(wrapper.containsMatchingElement(children)).toBe(true);
  });
});
