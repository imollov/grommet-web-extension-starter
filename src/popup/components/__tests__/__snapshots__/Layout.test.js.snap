// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component should match snapshot 1`] = `
<Grommet
  theme={
    Object {
      "button": Object {
        "extend": Array [
          [Function],
        ],
      },
      "global": Object {
        "colors": Object {
          "background": Object {
            "dark": "#000000",
            "light": "#ffffff",
          },
        },
        "font": Object {
          "family": "-apple-system,
         BlinkMacSystemFont, 
         \\"Segoe UI\\", 
         Roboto, 
         Oxygen, 
         Ubuntu, 
         Cantarell, 
         \\"Fira Sans\\", 
         \\"Droid Sans\\",  
         \\"Helvetica Neue\\", 
         Arial, sans-serif,  
         \\"Apple Color Emoji\\", 
         \\"Segoe UI Emoji\\", 
         \\"Segoe UI Symbol\\"",
        },
      },
    }
  }
>
  <Box
    height={
      Object {
        "min": "350px",
      }
    }
    width={
      Object {
        "min": "350px",
      }
    }
  >
    <div>
      Test
    </div>
  </Box>
</Grommet>
`;
